### PR TITLE
Implement GET api for all requests within 3 months

### PR DESCRIPTION
**Pull request by khangnh107**
### Overall:
- Add GET api endpoint for getting all requests created within 3 months.
- Added created_date field to Request's schema.

### File Changes:
- server/controllers/RequestController.js
- server/routers/RequestRouter.js
- server/models/Request.js

### Test:
Screenshot 2024-04-07 at 7 08 25 PM

### CODE REVIEW

File: server/controllers/RequestController.js [hunk 1]
```diff
@@ -17,6 +17,39 @@ export const getOneRequest = async (req, res) => {
...
+
+    for (startMonth = 0; startMonth <= 9; startMonth += 3) {
+      endMonth = startMonth + 2;
+      if (startMonth <= thisMonth && thisMonth <= endMonth) {
+        break;
+      }
+    }
+
+    const startDate = new Date(thisYear, startMonth, 1);
+    const endDate = new Date(new Date(thisYear, endMonth + 1, 1) - 1);
```
Comment (hoangr1010) [inline server/controllers/RequestController.js:36]: Can you have some in-line comment to make it understandable?

File: server/controllers/RequestController.js [hunk 2]
```diff
@@ -17,6 +17,39 @@ export const getOneRequest = async (req, res) => {
...
+
+    const { candidate_id } = req.body;
+    const requests = await Request.find({candidate_id: candidate_id, created_date: {$gte: startDate, $lte: endDate}});
```
Comment (hoangr1010) [inline server/controllers/RequestController.js:39]: indent nested object

File: server/models/Request.js [hunk 3]
```diff
@@ -32,6 +32,11 @@ const requestSchema = new Schema({
     type: Number,
     required: true,
   },
+  created_date: {
+    type: Date,
+    default: Date.now,
+    required: true,
+  }
```
Comment (hoangr1010) [inline server/models/Request.js:39]: you can replace all this by { timestamps: true }, mongodb will automatically add createdAt field to a document.
Screenshot 2024-04-07 at 8 49 33 PM

**Review by hoangr1010 — CHANGES_REQUESTED** [review]
Some comments and change requests

File: server/models/Request.js (hunk 3 above)
Comment (khangnh107) [reply server/models/Request.js:39]: Done

File: server/controllers/RequestController.js (hunk 2 above)
Comment (khangnh107) [reply server/controllers/RequestController.js]: Done

File: server/controllers/RequestController.js (hunk 1 above)
Comment (khangnh107) [reply server/controllers/RequestController.js]: Done

File: server/models/Request.js (hunk 3 above)
Comment (khangnh107) [reply server/models/Request.js]: Done

File: server/controllers/RequestController.js [hunk 4]
```diff
@@ -17,6 +17,52 @@ export const getOneRequest = async (req, res) => {
...
+      endMonth = startMonth + 2;
+      if (startMonth <= thisMonth && thisMonth <= endMonth) {
+        break;
+      }
+    }
+
+    // Get the specific start date and end date from the months gotten in the 3-month period
+    // For example, start month is April and end month is June, so the start date is April 1st and the end date is June 30th
+    const startDate = new Date(thisYear, startMonth, 1);
+    const endDate = new Date(new Date(thisYear, endMonth + 1, 1) - 1);
```
Comment (hoangr1010) [inline server/controllers/RequestController.js:42]: I think you can refactor this into a function that take a current date, and return startDate and endDate

**Review by hoangr1010 — CHANGES_REQUESTED** [review]
one more comment hehe

File: server/controllers/RequestController.js (hunk 4 above)
Comment (khangnh107) [reply server/controllers/RequestController.js]: Done

**Review by hoangr1010 — APPROVED** [review]
LGTM!

**Review by lnguyen2693 — APPROVED** [review]

**Pull request merged**